api/authentication.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework import exceptions
+from rest_framework.authentication import TokenAuthentication, get_authorization_header
+from api.models import Token, IP
+
+
+class TokenIPAuth(TokenAuthentication):
+    model = Token
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+
+        if not auth or auth[0].lower() != self.keyword.lower().encode():
+            return None
+
+        if len(auth) == 1:
+            msg = _('Invalid token header. No credentials provided.')
+            raise exceptions.AuthenticationFailed(msg)
+        elif len(auth) > 2:
+            msg = _('Invalid token header. Token string should not contain spaces.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        try:
+            key = auth[1].decode()
+        except UnicodeError:
+            msg = _('Invalid token header. Token string should not contain invalid characters.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        model = self.get_model()
```

Comment:
I think it's better to have a separate method which authenticates a key for a certain ip like ``authenticate_credentials`` in the super class. The only difference is that you should also pass the ip instead of just the token.

To me it looks like you could just override ``authenticate_credentials``. You just copied the ``authenticate`` method source code and instead of calling ``authenticate_credentials`` you started processing here.